### PR TITLE
Feature/ontogene

### PR DESCRIPTION
**Pull request by PhilippKuntschik**

### CODE REVIEW

File: src/python/weblyzard_api/client/ontogene/__init__.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+INTERNAL_PROFILE_PREFIX = 'extras.'
+LOGGER = logging.getLogger('weblyzard_api.client.ontogene.oger')
+
+DEFAULT_MAX_RETRY_DELAY = 15
+DEFAULT_MAX_RETRY_ATTEMPTS = 5
+DAYS_BACK_DEFAULT = 20
+
+#GET URL/{pubmed|pmd}/{txt|bioc|pxml|nxml|pxml.gz}/DOC_ID
+
+class SpecialGetRequest(object):
```
Comment (maxgoebel) [inline src/python/weblyzard_api/client/ontogene/__init__.py]: this seems to not be used? If not, please remove

File: src/python/weblyzard_api/client/ontogene/__init__.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    def __init__(self, url=OGER_API_URL, default_timeout=None):
+        '''
+        :param url: URL of the jeremia web service
+        #:param usr: optional user name
+        #:param pwd: optional password
+        '''
+        #no usr=WEBLYZARD_API_USER, pwd=WEBLYZARD_API_PASS,
+        ##user=usr, password=pwd,
+        MultiRESTClient.__init__(self, service_urls=url, default_timeout=default_timeout)
+        self.profile_cache = []
```
Comment (maxgoebel) [inline src/python/weblyzard_api/client/ontogene/__init__.py]: what is this for? If not used, please remove

File: src/python/weblyzard_api/client/ontogene/__init__.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        :returns: OGER document converted to Recognyze format.
+        '''
+        res = oger_json
+        
+        entities = {}
+        d2 = []
+        
+        #TODO: talk to OGER developers to fix this rather than have it fixed here!
+        #this is a fix for offset which always seems to be delayed with this value
+        offset = res['documents'][0]['passages'][1]['annotations'][0]['locations'][0]['offset']
```
Comment (maxgoebel) [inline src/python/weblyzard_api/client/ontogene/__init__.py]: is this guaranteed to exist? could do with some exception handling IMO

File: src/python/weblyzard_api/client/ontogene/__init__.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        
+        for passage in res['documents'][0]['passages']:
+        
+            for rs in passage['annotations']:
+                print(rs)
```
Comment (maxgoebel) [inline src/python/weblyzard_api/client/ontogene/__init__.py]: please remove all prints...

File: src/python/weblyzard_api/client/ontogene/__init__.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                    "start": start - offset,
+                    "end": end - offset,
+                    "offset": offset,
+                    "confidence": 1,
+                    "preferred_name": rs['infons']['preferred_form'],
+                    "entity_type": rs['infons']['type']
+                }
+                #print(ditem)         
+                d2.append(ditem)
+        print(d2)
```
Comment (maxgoebel) [inline src/python/weblyzard_api/client/ontogene/__init__.py]: remove

File: src/python/weblyzard_api/client/ontogene/__init__.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        url = OGER_API_URL + "upload/txt/bioc_json"
+        files = {'file1': (docid, doctext)}
+        
+        r = requests.post(url, files=files)
+        #print(r.text)
+        res = r.json()
+        return self.convert_document(res)
+ 
+    
+    def search_documents(self):
```
Comment (maxgoebel) [inline src/python/weblyzard_api/client/ontogene/__init__.py]: please remove if not used

File: src/python/weblyzard_api/client/ontogene/__init__.py [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        
+        for passage in res['documents'][0]['passages']:
+        
+            for rs in passage['annotations']:
+                print(rs)
+                start = rs['locations'][0]['offset']
+                end = rs['locations'][0]['offset'] + len(rs['text'])
+                ditem = {
+                    "key": rs['infons']['original_id'],
+                    "resource": rs['infons']['original_resource'],
```
Comment (maxgoebel) [inline src/python/weblyzard_api/client/ontogene/__init__.py]: what information is `resource` meant to hold, this field does not exist in the [webLyzard Annotation model ](https://github.com/weblyzard/weblyzard_api/blob/master/src/python/weblyzard_api/model/__init__.py#L93). I recomment to stick to the model, if possible

File: src/python/weblyzard_api/client/ontogene/__init__.py [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    '''
+
+    def convert_document(self, oger_json):
+        '''
+        :returns: OGER document converted to Recognyze format.
+        '''
+        res = oger_json
+        
+        entities = {}
+        d2 = []
```
Comment (maxgoebel) [inline src/python/weblyzard_api/client/ontogene/__init__.py]: rename to `result`

File: src/python/weblyzard_api/client/ontogene/__init__.py (hunk 2 above)
Comment (adib2011) [reply src/python/weblyzard_api/client/ontogene/__init__.py]: removed

File: src/python/weblyzard_api/client/ontogene/__init__.py (hunk 3 above)
Comment (adib2011) [reply src/python/weblyzard_api/client/ontogene/__init__.py]: added error handling

**Pull request merged**